stub_uploader/build_wheel.py

```diff
@@ -307,18 +301,13 @@ def collect_package_data(base_path: Path) -> PackageData:
             continue
         elif entry.is_file() and entry.suffix == ".pyi":
             pkg_name = entry.stem
-            # Module -> package transformation is done while copying.
-            stub_files = ["__init__.pyi"]
         elif entry.is_dir():
             pkg_name = entry.name
-            stub_files = find_stub_files(str(entry))
+            validate_only_stub_files(str(entry))
         else:
             raise ValueError(f"Only stub files are allowed, not {entry.name!r}")
-        package_data[pkg_name + SUFFIX] = [*stub_files, META]
-    pkg_data = PackageData(base_path, package_data)
-    for package in pkg_data.top_level_non_namespace_packages:
-        pkg_data.add_file(package, "py.typed")
```

Comment:
As far as I can see, this is the only call of `PackageData.add_file()`, so we can remove that method.